modules/ftp.go

```diff
@@ -0,0 +1,189 @@
+package modules
+
+import (
+	"net"
+	"regexp"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/zmap/zgrab2"
+)
+
+// FTPScanResults is the output of the scan. Identical to the original from zgrab, with the addition of TLSLog.
+type FTPScanResults struct {
+	Banner      string         `json:"banner,omitempty"`
+	AuthTLSResp string         `json:"auth_tls_resp,omitempty"`
+	AuthSSLResp string         `json:"auth_ssl_resp,omitempty"`
+	TLSLog      *zgrab2.TLSLog `json:"tls,omitempty"`
+}
+
+// FTP-specific command-line flags. Taken from the original zgrab (TODO: should FTPAuthTLS be on by default?).
+type FTPFlags struct {
+	zgrab2.BaseFlags
+	zgrab2.TLSFlags
+	Verbose    bool `long:"verbose" description:"More verbose logging, include debug fields in the scan results"`
```

Comment:
This should this be a module-level flag, or global?

Reply:
The SSH module I've been using as my template had it as a module-level flag, so I've followed that everywhere else.

I could see some value in having detailed logs for certain modules but not others, but I don't know if that outweighs the potential confusion on the user side ("I said verbose, why isn't it working everywhere?")

Reply:
I kind of like the idea of having the module being able to mark certain fields to only be output in verbose mode (e.g., using a Go tag on the output field), and then having the framework then output correctly based on the framework flag and the tag on the output structure. @dadrian, thoughts on this one?

Reply:
We started that concept with the zgrab:"debug" tag (currently, for some of the verbose mysql fields, and the TLS client bits), but the custom marshaller to omit them isn't implemented yet (https://github.com/zmap/zgrab2/issues/29).

Reply:
Maybe it makes sense to knock that out now if we're going to use it in a bunch of places? Then we can just take advantage of that everywhere.

Reply:
I agree it should be framework level. I'm not sure we need to implement it next, though. There's no particular reason we can't just have overly verbose output all the time by default while ZGrab2 is still under development. We just need it deployed before we scan.